aiohttp/client_reqrep.py

```diff
@@ -1011,36 +1006,33 @@ def get_encoding(self) -> str:
         mimetype = helpers.parse_mimetype(ctype)
 
         encoding = mimetype.parameters.get("charset")
-        if encoding:
-            try:
-                codecs.lookup(encoding)
-            except LookupError:
-                encoding = None
-        if not encoding:
-            if mimetype.type == "application" and (
-                mimetype.subtype == "json" or mimetype.subtype == "rdap"
-            ):
-                # RFC 7159 states that the default encoding is UTF-8.
-                # RFC 7483 defines application/rdap+json
-                encoding = "utf-8"
-            elif self._body is None:
-                raise RuntimeError(
-                    "Cannot guess the encoding of " "a not yet read body"
-                )
-            else:
-                encoding = chardet.detect(self._body)["encoding"]
+
         if not encoding:
-            encoding = "utf-8"
+            return "utf-8"
 
-        return encoding
+        try:
+            return codecs.lookup(encoding).name
+        except LookupError:
+            return "utf-8"
 
     async def text(self, encoding: Optional[str] = None, errors: str = "strict") -> str:
         """Read response payload and decode."""
         if self._body is None:
             await self.read()
 
-        if encoding is None:
-            encoding = self.get_encoding()
+        # If the user specified the encoding, do not attempt to guess the
+        # encoding. May raise UnicodeDecodeError if user specified an invalid
+        # encoding
+        if encoding is not None:
+            return self._body.decode(encoding, errors=errors)  # type: ignore[union-attr]
+
+        try:
+            return self._body.decode(self.get_encoding(), errors=errors)  # type: ignore[union-attr]
```

Comment:
The old code achieves the same thing here without code duplication.

```suggestion
        if encoding is None:
            encoding = self.get_encoding()

        try:
            return self._body.decode(encoding, errors=errors)  # type: ignore[union-attr]
```